connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaConfigBackingStoreMockitoTest.java

```diff
@@ -347,6 +380,165 @@ public void testPutConnectorConfigWithTargetState() throws Exception {
         verify(configLog).stop();
     }
 
+    @Test
+    public void testPutConnectorConfigProducerError() throws Exception {
+        expectStart(Collections.emptyList(), Collections.emptyMap());
+        expectPartitionCount(1);
+
+        configStorage.setupAndCreateKafkaBasedLog(TOPIC, config);
+        verifyConfigure();
+        configStorage.start();
+
+        when(converter.fromConnectData(TOPIC, KafkaConfigBackingStore.CONNECTOR_CONFIGURATION_V0, CONNECTOR_CONFIG_STRUCTS.get(0)))
+                .thenReturn(CONFIGS_SERIALIZED.get(0));
+        when(configLog.sendWithReceipt(anyString(), any(byte[].class))).thenReturn(producerFuture);
+
+        // Verify initial state
+        ClusterConfigState configState = configStorage.snapshot();
+        assertEquals(-1, configState.offset());
+        assertEquals(0, configState.connectors().size());
+
+        when(producerFuture.get(anyLong(), any(TimeUnit.class))).thenThrow(
+                new ExecutionException(new TopicAuthorizationException(Collections.singleton("test"))));
+
+        // verify that the producer exception from KafkaBasedLog::send is propagated
+        ConnectException e = assertThrows(ConnectException.class, () -> configStorage.putConnectorConfig(CONNECTOR_IDS.get(0),
+                SAMPLE_CONFIGS.get(0), null));
+        assertTrue(e.getMessage().contains("Error writing connector configuration to Kafka"));
```

Comment:
Could we verify the `e.getCause()` to make sure the error is caused by what we expect?